lib/AST/ASTContext.cpp

```diff
@@ -4421,6 +4422,11 @@ CanTypeWrapper<OpenedArchetypeType> OpenedArchetypeType::getNew(
     GenericEnvironment *environment, Type interfaceType,
     ArrayRef<ProtocolDecl *> conformsTo, Type superclass,
     LayoutConstraint layout) {
+  // FIXME: It'd be great if all of our callers could submit interface types.
```

Comment:
Callers should be able to provide a substitution map to map generic parameters to archetypes or type variables.